src/NLog/Logger.cs

```diff
@@ -94,17 +97,70 @@ public bool IsEnabled(LogLevel level)
             return GetTargetsForLevel(level) != null;
         }
 
+        /// <summary>
+        /// Creates new logger that automatically appends the specified property to all log events (without changing current logger)
+        /// </summary>
+        /// <param name="propertyKey">Property Name</param>
+        /// <param name="propertyValue">Property Value</param>
+        /// <returns>New Logger object that automatically appends specified property</returns>
+        public Logger WithProperty(string propertyKey, object propertyValue)
+        {
+            if (string.IsNullOrEmpty(propertyKey))
+                throw new ArgumentException(nameof(propertyKey));
+
+            Logger newLogger = Factory.CreateNewLogger(GetType()) ?? new Logger();
+            newLogger.Initialize(Name, _configuration, Factory);
+            var contextProperties = _contextProperties != null
+                ? new Dictionary<string, object>(_contextProperties)
+                : new Dictionary<string, object>();
+            contextProperties[propertyKey] = propertyValue;
+            newLogger._contextProperties = contextProperties;
+            newLogger._contextLogger = _contextLogger;
+            newLogger._isTraceEnabled = true;
+            newLogger._isDebugEnabled = true;
+            newLogger._isInfoEnabled = true;
+            newLogger._isWarnEnabled = true;
+            newLogger._isErrorEnabled = true;
+            newLogger._isFatalEnabled = true;
```

Comment:
Why is everything enabled for the new logger?

Reply:
Noise from my first attempt. Will remove it